fractal_server/tasks/collection.py

```diff
@@ -482,6 +487,50 @@ async def _pip_install(
     await execute_command(
         cwd=venv_path, command=cmd_install, logger_name=logger_name
     )
+    if task_pkg.pinned_package_versions:
+        for (
+            pinned_pkg_name,
+            pinned_pkg_version,
+        ) in task_pkg.pinned_package_versions.items():
+
+            logger.debug(
+                "Specific version required: "
+                f"{pinned_pkg_name}=={pinned_pkg_version}"
+            )
+            stdout_inspect = await execute_command(
+                cwd=venv_path,
+                command=f"{pip} show {pinned_pkg_name}",
+                logger_name=logger_name,
+            )
+            current_version = next(
+                line.split()[-1]
+                for line in stdout_inspect.split("\n")
+                if line.startswith("Version:")
+            )
+            logger.debug(
```

Comment:
This log should go up, before the pip-show command. Otherwise it will not be visible in case pip-show fails